tools/hangar/__snapshots__/test_corpus/valid/extern_implementation.w.ts.snap

```diff
@@ -7,36 +7,28 @@ constructor({  }) {
 
 }
 static async regex_inflight(pattern, text)  {
-	return require(\\"../extern/8b481fc062bdfc1c35bbcbbab32dcf0805e52e773caa1c61121b09e2db8a0b5b.js\\")[\\"regex_inflight\\"](pattern, text)
+	return require(\\"../extern/09e024cf966860b34adf9cbfd8a494763760f871b1f5c00bbc98853b9989078e.js\\")[\\"regex_inflight\\"](pattern, text)
 }
 static async get_uuid()  {
-	return require(\\"../extern/8b481fc062bdfc1c35bbcbbab32dcf0805e52e773caa1c61121b09e2db8a0b5b.js\\")[\\"get_uuid\\"]()
+	return require(\\"../extern/09e024cf966860b34adf9cbfd8a494763760f871b1f5c00bbc98853b9989078e.js\\")[\\"get_uuid\\"]()
+}
+static async get_data()  {
+	return require(\\"../extern/09e024cf966860b34adf9cbfd8a494763760f871b1f5c00bbc98853b9989078e.js\\")[\\"get_data\\"]()
 }
 async call()  {
 	{
   {((cond) => {if (!cond) throw new Error(\`assertion failed: '(await Foo.regex_inflight(\\"[a-z]+-\\\\\\\\d+\\",\\"abc-123\\"))'\`)})((await Foo.regex_inflight(\\"[a-z]+-\\\\\\\\d+\\",\\"abc-123\\")))};
   const uuid = (await Foo.get_uuid());
   {((cond) => {if (!cond) throw new Error(\`assertion failed: '(uuid.length === 36)'\`)})((uuid.length === 36))};
+  {((cond) => {if (!cond) throw new Error(\`assertion failed: '((await Foo.get_data()) === \\"Cool data!\\")'\`)})(((await Foo.get_data()) === \\"Cool data!\\"))};
 }
 }}
 exports.Foo = Foo;"
 `;
 
-exports[`wing compile -t tf-aws > extern/8b481fc062bdfc1c35bbcbbab32dcf0805e52e773caa1c61121b09e2db8a0b5b.js 1`] = `
-"exports.get_greeting = function(name) {
-  return \`Hello, \${name}!\`;
-};
-
-exports.regex_inflight = async function(pattern, text) {
-  const regex = new RegExp(pattern);
-  return regex.test(text);
-};
+exports[`wing compile -t tf-aws > extern/09e024cf966860b34adf9cbfd8a494763760f871b1f5c00bbc98853b9989078e.js 1`] = `"module.exports = require(require.resolve(\`./external_js.js\`, {paths: [process.env.WING_FILE_DIR]}));"`;
```

Comment:
Maybe we can avoid this level of indirection some way...

Reply:
Done. Good callout, I was too busy changing a small piece of the functionality I didn't consider just deleting 80% of it!